client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,61 @@ def build_template(
-                keep_placeholders,
-                create_first_version,
-            ]
+
+        # Get default values if not provided
+        if (
+                template_path is None
+                or keep_placeholders is None
+                or create_first_version is None
         ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
             if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
+                keep_placeholders: bool = preset["keep_placeholder"]
             if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
-
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
+                create_first_version: bool = preset["create_first_version"]
 
-        # if first version is created, import template
-        # and populate placeholders
+        # Build the template if current workfile is a new unsaved file
+        # (that's detected by checking if it returns any current filepath)
         if (
-            create_first_version
-            and workfile_creation_enabled
-            and created_version_workfile
+                # If not a workfile creation, an explicit load template
+                # was requested, so we always want to build the template
+                not workfile_creation_enabled
+                # Or if workfile creation, but we're not in an active file
+                # we still need to build the "new workfile template"
+                or not self.host.get_current_workfile()
         ):
+            self.log.info(f"Building the workfile template: {template_path}")
             self.import_template(template_path)
             self.populate_scene_placeholders(
                 level_limit, keep_placeholders)
 
-            # save workfile after template is populated
-            self.save_workfile(created_version_workfile)
+        if not workfile_creation_enabled:
+            # Do not consider saving a first workfile version, if this
+            # is not set to be a "workfile creation". Then we assume
+            # only the template should get build.
+            return
 
-        # ignore process if first workfile is enabled
-        # but a version is already created
-        if workfile_creation_enabled:
+        if not create_first_version:
+            # Do not save whatsoever
             return
 
-        self.import_template(template_path)
-        self.populate_scene_placeholders(
-            level_limit, keep_placeholders)
+        # If there is no existing workfile, save the first version
```

Comment:
where are you checking if there is no existing workfile? just reading the function `get_first_workfile_version_to_create` that's not obvious

Reply:
This happens inside `get_first_workfile_version_to_create`
the way it works: 

If my task doesn't have any workfiles then this environment variable `AYON_LAST_WORKFILE` will always refer to the first version of the workfile. 
Then, we just use `os.path.exists(last_workfile_path)` to check if the file exists. 

The only caveat with the logic above is that the mentioned env variable is not updated on context change.
This may result in an expected behavior.. but currently, we are aware of this potential issue but didn't debug it nor having any reproducible steps yet.

![image](https://github.com/user-attachments/assets/bb5b6976-7910-4abd-a6c6-eef41fd60dc0)

Reply:
yeah I know how the code works but I was talking more about how readable the code and its corresponding docstring look. It's not too bad but to me a function that is called `get_first_workfile_version_to_create` and returns a variable "workfile_path" is hard to associate it to a boolean. I think it's the name of the function and its logic is confusing from an API point of view. But perhaps we just need to expand the docstring and say something like `# the function returns False if a workfile already exists`